iml-agent/src/daemon_plugins/ntp.rs

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::ImlAgentError,
+    cmd::cmd_output,
+    daemon_plugins::{as_output, DaemonPlugin, Output},
+    systemd,
+};
+use futures::{future, Future, Stream};
+use iml_wire_types::ntp::{TimeOffset, TimeStatus, TimeSync};
+use lazy_static::lazy_static;
+use regex::Regex;
+use std::path::Path;
+use std::process;
+use tracing::info;
+
+static NTP_CONFIG_FILE: &'static str = "/etc/ntp.conf";
+
+#[derive(Debug)]
+pub struct Ntp;
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct SystemdUnit {
+    enabled: bool,
+    active: bool,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct TimeSyncServices {
+    ntpd: SystemdUnit,
+    chronyd: SystemdUnit,
+}
+
+impl TimeSyncServices {
+    fn is_ntp_only(&self) -> bool {
+        self.ntpd.enabled && self.ntpd.active && !self.chronyd.enabled && !self.chronyd.active
+    }
+}
+
+fn is_ntp_synced(output: &str) -> TimeSync {
+    let x = match output {
+        "b true" => TimeSync::Synced,
+        "b false" => TimeSync::Unsynced,
+        _ => TimeSync::Unsynced,
+    };
+
+    info!("is_ntp_synced: {:?}; equal? {}", x, output == "b true");
```

Comment:
This should be `debug`. We don't want a message for each tick.

Reply:
agreed, I will update all of my comments to use a debug level.